core/src/main/java/inetsoft/report/composition/WorksheetEngine.java

```diff
@@ -199,7 +198,14 @@ protected String createTemporarySheetId(AssetEntry entry, RuntimeSheet sheet,
    {
       String id = getNextID(entry, user);
       sheet.setID(id);
-      amap.put(id, sheet);
+
+      try {
+         amap.putSheet(id, sheet).get();
```

Comment:
Need to specify a reasonable timeout for Future.get(). If there is a communication failure or similar, this can lock a thread.